function login(email, password) {
    return fetch('/login', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
        },
        body: JSON.stringify({ email, password }),
    })
    .then(async response => {
        if (!response.ok) {
            const error = await parseError(response);
            console.error(error);
            throw new Error(error);
        } else {
            return {
                success: true,
                data: await response.json(),
            }
        }
    })
    .catch(error => {
        console.error('Error:', error);
        return {
            success: false,
            error: error,
        };
    });
}

function register(email, password) {
    return fetch('/register', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
        },
        body: JSON.stringify({ 
            email, 
            password,
        }),
    })
    .then(async response => {
        if (!response.ok) {
            const error = await parseError(response);
            console.error(error);
            throw new Error(error);
        } else {
            return {
                success: true,
                data: await response.json(),
            }
        }
    })
    .catch(error => {
        console.log(error?.error);
        console.error('Error:', error);
        return {
            success: false,
            error: error,
        };
    });
}

async function parseError(error) {
    try {
        const text = await error.text();
        const json = JSON.parse(text);
        return json.error;
    } catch (error) {
        console.error('Error:', error);
        return error;
    }
}

async function logout() {
    try {
        const response = await fetch('/logout', {
            method: 'DELETE',
            headers: {
                'Content-Type': 'application/json',
            },
        })
        if (!response.ok) {
            const error = await parseError(response);
            console.error(error);
            throw new Error(error);
        } else {
            return {
                success: true,
                data: await response.json(),
            }
        }
    } catch (error) {
        console.error('Error:', error);
        return {
            success: false,
            error: error,
        };
    }
}